lib/read_only/generic/README.md

```diff
@@ -0,0 +1,90 @@
+# Read-only Verification Library for Generic Fortran Code
+
+This library implements the [PSyData API](
+https://psyclone.readthedocs.io/en/latest/psy_data.html#read-only-verification-library-for-gocean)
+to verify that variables declared read-only are not modified (overwritten) in
+a kernel.
+
+A full runnable example can be found in [``examples/nemo/eg6/``](
+https://github.com/stfc/PSyclone/tree/master/examples/nemo/eg6).
+
+## Dependencies
+
+This library uses the [PSyData API](
+https://psyclone.readthedocs.io/en/stable/psy_data.html) to interface
+with the application. It needs the ReadOnly (``read_only_base.jinja``) and
+PSyData (``psy_data_base.jinja``) base classes, which are included in PSyclone
+installation. These Jinja templates are processed to create
+the generic read-only verification code for integer and real variables (both
+32- and 64-bit), logical and character variables. It will support scalar, and
+1- to 4-dimensional arrays. The ``read_only.f90`` file contains no actual
+executable code, it inherits from the base classes, and is only used so that
+the expected module- and type-names are available for PSyclone.
+
+## Compilation
+
+The library is compiled with ``make`` using the provided ``Makefile``. The
+environment variables ``$F90`` and ``$F90FLAGS`` can be set to point to the
+[Fortran compiler](./../../README.md#compilation) and flags to use. They
+default to ``gfortran`` and the empty string.
+
+The locations of the ReadOnly and PSyData base classes are specified
+using the environment variables ``$LIB_TMPLT_DIR`` and ``$PSYDATA_LIB_DIR``,
+respectively. They default to the relative paths to the
+[``lib/read_only``](./../) and top-level [``lib``](./../../) directories.
+
+The compilation process will create the wrapper library ``lib_read_only.a``.
+
+Similar to compilation of the [examples](
+https://psyclone.readthedocs.io/en/latest/examples.html#compilation), the
```

Comment:
Should be `tutorials_and_examples.html` I think.

Reply:
Indeed - that's broken in all three versions of the library  - oops, it's actually broken in all READMEs in the lib directory. All fixed now.